src/push/Permission.js

```diff
@@ -41,13 +41,17 @@ export default class Permission {
     _requestWithCallback(onGranted: () => void, onDenied: () => void) {
         const existing = this.get();
 
+        var resolved = false;
```

Comment:
I added this (and reference it in the body of the `resolve` function to prevent doubly-executing the resolution callback. This will prevent a future issue if Safari (or Chrome) support both the callback and Promise style of `requestPermission` simultaneously.

Reply:
I highly doubt that would happen. But being future-proof can't hurt. I consider this resolved.